rmw_connextdds_common/src/common/rmw_info.cpp

```diff
@@ -235,19 +235,6 @@ rmw_api_connextdds_count_publishers(
   RMW_CHECK_ARGUMENT_FOR_NULL(topic_name, RMW_RET_INVALID_ARGUMENT);
   RMW_CHECK_ARGUMENT_FOR_NULL(count, RMW_RET_INVALID_ARGUMENT);
 
-  int validation_result = RMW_TOPIC_VALID;
-  rmw_ret_t ret =
-    rmw_validate_full_topic_name(topic_name, &validation_result, nullptr);
-  if (RMW_RET_OK != ret) {
-    return ret;
-  }
-  if (RMW_TOPIC_VALID != validation_result) {
-    const char * reason =
-      rmw_full_topic_name_validation_result_string(validation_result);
-    RMW_CONNEXT_LOG_ERROR_A_SET("invalid topic name: %s", reason)
-    return RMW_RET_INVALID_ARGUMENT;
-  }
```

Comment:
why is this change required?

Reply:
never mind, I see it was restored here.
Though it wasn't restored in `rmw_api_connextdds_get_subscriptions_info_by_topic()` `rmw_api_connextdds_get_publishers_info_by_topic()`

Reply:
I had to remove the check from `rmw_api_connextdds_get_*_info_by_topic()` because otherwise test `TestNodeGraph.get_info_by_topic` would fail:

```
C++ exception with description "Failed to get information by topic for publishers:invalid topic name: topic name must be absolute, it must lead with a '/', at /home/jenkins-agent/workspace/ci_linux/ws/src/ros2/rmw_connextdds/rmw_connextdds_common/src/common/rmw_info.cpp:550, at /home/jenkins-agent/workspace/ci_linux/ws/src/ros2/rcl/rcl/src/rcl/graph.c:677: error not set" thrown in the test body.
```

(from [your CI run for #25](https://ci.ros2.org/job/ci_linux/14302/testReport/rclcpp/TestNodeGraph/get_info_by_topic/))

Reply:
Also, I think it was the same test that required the propagation of `history`, but I have to double check.

Reply:
> I had to remove the check from rmw_api_connextdds_get_*_info_by_topic() because otherwise test TestNodeGraph.get_info_by_topic would fail:

The validation should only be applied when `no_mangle` is `false`, and skipped when `true`.
I'm almost sure that's why the test was failing, though I haven't tested.

I don't mind if the validation gets completely removed though.

Reply:
I can restore the logic for `!no_mangle`. I'd rather catch it now with these changeset than leaving it for later.

Reply:
Sounds good to me

Reply:
I don't quite understand the test, but restoring the validation checks only for `!no_mangle` makes [this test](https://github.com/ros2/rclcpp/blob/master/rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp#L446) fail:

```
[ RUN      ] TestNodeGraph.get_info_by_topic_rcl_remap_topic_name_nullptr
RTI Data Distribution Service Non-commercial license is for academic, research, evaluation and personal use only. USE FOR COMMERCIAL PURPOSES IS PROHIBITED. See RTI_LICENSE.TXT for terms. Download free tools at rti.com/ncl. License issued to Non-Commercial User license@rti.com For non-production use only.
Expires on 00-jan-00 See www.rti.com for more information.
[ERROR] [1618004974.610929217] [rmw_connextdds]: invalid topic name: topic name must not be empty
/home/asorbini/workspace-rmw/issues/21/ws_ros-src/src/ros2/rclcpp/rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp:446: Failure
Expected: node_graph()->get_publishers_info_by_topic("topic", false) doesn't throw an exception.
  Actual: it throws.
[  FAILED  ] TestNodeGraph.get_info_by_topic_rcl_remap_topic_name_nullptr (329 ms)
```

The complaint from `rmw_connextdds` seems legitimate to me, but somehow the test doesn't expect an exception.

Reply:
@ivanpauno @hidmic any input on this issue?

Can you confirm if my assumption that the test should in fact expect an exception because the topic name is remapped to an empty name is correct?

If so, I could reintroduce the checks in `rmw_api_connextdds_get_subscriptions_info_by_topic()` `rmw_api_connextdds_get_publishers_info_by_topic()`, but other RMW implementation will likely need to be updated.

I can also open a PR to fix the test and make reintroduce the checks in `rmw_connextdds` in a separate PR.

If, on the other hand, the test is right, then I guess the code is good "as is".

Reply:
TL; DR let's leave the checks out.

So, the test itself is odd. Its name suggests it is mocking to force a [no matching rule case](https://github.com/ros2/rcl/blob/414fdbd479b83f2b8af8235ad1fa3aeba2b0027f/rcl/include/rcl/remap.h#L101-L103) but then yields an empty string, which is an invalid remapping (and impossible to construct, for that matter). That gets forwarded to the `rmw` implementation, where the graph query yields nothing. They could be doing these checks, all the information is there, but alas, the `rmw` API contract [does not impose any restriction](https://github.com/ros2/rmw/blob/a6420837ccfb66b9df8addc7c1dadd50f39ec74a/rmw/include/rmw/get_topic_endpoint_info.h#L26-L89) (an omission, most likely). 

For the time being, I think the best move is to stay consistent with the API contract and not do any checks. We can improve the test and extend the API contract for H-Turtle. Sorry for the back and forth @asorbini.

Reply:
Thank you for the explanation @hidmic, I agree that leaving the checks out is the best solution for now.